assets/js/base/components/cart-checkout/totals/discount/index.js

```diff
@@ -53,34 +57,48 @@ const TotalsDiscount = ( {
 						showSpinner={ false }
 					>
 						<ul className="wc-block-components-totals-discount__coupon-list">
-							{ cartCoupons.map( ( cartCoupon ) => (
-								<RemovableChip
-									key={ 'coupon-' + cartCoupon.code }
-									className="wc-block-components-totals-discount__coupon-list-item"
-									text={ cartCoupon.code }
-									screenReaderText={ sprintf(
-										/* translators: %s Coupon code. */
-										__(
-											'Coupon: %s',
-											'woo-gutenberg-products-block'
-										),
-										cartCoupon.code
-									) }
-									disabled={ isRemovingCoupon }
-									onRemove={ () => {
-										removeCoupon( cartCoupon.code );
-									} }
-									radius="large"
-									ariaLabel={ sprintf(
-										/* translators: %s is a coupon code. */
-										__(
-											'Remove coupon "%s"',
-											'woo-gutenberg-products-block'
-										),
-										cartCoupon.code
-									) }
-								/>
-							) ) }
+							{ cartCoupons.map( ( cartCoupon ) => {
+								const filteredCouponCode = __experimentalApplyCheckoutFilter(
+									{
+										validation: mustBeString,
+										arg: {
+											context: 'summary',
+											coupon: cartCoupon,
+										},
```

Comment:
I will add cart data and extensions data here in https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4164 but we would need them in all filters going forward.